tmk_core/protocol/chibios/chibios.c

```diff
@@ -157,32 +180,25 @@ void protocol_post_init(void) {
 }
 
 void protocol_pre_task(void) {
+    usb_event_queue_task();
 #if !defined(NO_USB_STARTUP_CHECK)
     if (USB_DRIVER.state == USB_SUSPENDED) {
         dprintln("suspending keyboard");
         while (USB_DRIVER.state == USB_SUSPENDED) {
             suspend_power_down();
```

Comment:
Maybe not the most important suggestion, but these comments used to exist (and might make applying the patch back a pain)
```suggestion
            /* Do this in the suspended state */
            suspend_power_down(); // on AVR this deep sleeps for 15ms
            /* Remote wakeup */
```